test/build-scheduler.js

```diff
@@ -0,0 +1,123 @@
+process.env.SQS_URL = "https://sqs.us-east-1.amazonaws.com/123abc/456def"
+process.env.ECS_CLUSTER = "ECS_CLUSTER"
+process.env.ECS_TASK = "federalist-builder"
+process.env.MAX_TASKS = 10
+process.env.PORT = 3000
+
+const expect = require("chai").expect
+const BuildScheduler = require("../src/build-scheduler")
+
+describe("BuildScheduler", () => {
+  it("it should run a task when a message is received from SQS and then delete the message", (done) => {
+    let hasReceivedMessage = false
+    const sqs = {}
+    sqs.receiveMessage = (params, callback) => {
+      if (!hasReceivedMessage) {
+        hasReceivedMessage = true
+        callback(null, {
+          Messages: [
+            {
+              Body: "{}"
+            }
+          ]
+        })
+      } else {
+        mockedSQSReceiveMessage(params, callback)
+      }
+    }
+
+    let hasDeletedMessage = false
+    sqs.deleteMessage = (params, callback) => {
+      hasDeletedMessage = true
+      mockedSQSDeleteMessage(params, callback)
+    }
+
+    let hasRunTask = false
+    const ecs = {}
+    ecs.runTask = (params, callback) => {
+      hasRunTask = true
+      mockedECSRunTask(params, callback)
+    }
+
+    const buildScheduler = new BuildScheduler()
+    mockECSServices(buildScheduler, { sqs, ecs })
+    buildScheduler.start()
+
+    setTimeout(() => {
```

Comment:
This can technically also be a `setImmediate` or `setTimeout(fn, 0)`, since what you want is to schedule the callback to run at the top of the next execution queue. This has the bonus of saving 250 milliseconds on the test. It also means you can remove the `stop` call (although perhaps that has other implications)